File: src/main/java/net/Chidoziealways/everythingjapanese/poi/ModPoiTypes.java
package net.Chidoziealways.everythingjapanese.poi;

import com.google.common.collect.ImmutableSet;
import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.Chidoziealways.everythingjapanese.block.ModBlocks;
import net.minecraft.world.entity.ai.village.poi.PoiType;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModPoiTypes {
    public static final DeferredRegister<PoiType> POI_TYPES = DeferredRegister.create(
            ForgeRegistries.POI_TYPES, EverythingJapanese.MOD_ID
    );

    public static final RegistryObject<PoiType> HELL_PORTAL = POI_TYPES.register("hell_portal",
            () -> new PoiType(ImmutableSet.copyOf(ModBlocks.HELL_PORTAL.get().getStateDefinition().getPossibleStates()), 1, 1));

    public static void register(IEventBus eventBus) {
        POI_TYPES.register(eventBus);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/poi/ModPoiTypes.kt
package net.Chidoziealways.everythingjapanese.poi

import com.google.common.collect.ImmutableSet
import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.Chidoziealways.everythingjapanese.block.JModBlocks
import net.minecraft.core.registries.Registries
import net.minecraft.world.entity.ai.village.poi.PoiType
import net.minecraft.world.level.block.state.BlockState
import net.neoforged.bus.api.IEventBus
import net.neoforged.neoforge.registries.DeferredRegister
import java.util.function.Supplier

object ModPoiTypes {
    val POI_TYPES = DeferredRegister.create(
        Registries.POINT_OF_INTEREST_TYPE, JAPANESE_MOD_ID
    )

    val HELL_PORTAL = POI_TYPES.register(
        "hell_portal",
        Supplier {
            PoiType(
                ImmutableSet.copyOf<BlockState>(
                    JModBlocks.HELL_PORTAL.getStateDefinition().getPossibleStates()
                ), 1, 1
            )
        })

    fun register(eventBus: IEventBus) {
        POI_TYPES.register(eventBus)
    }
}
